### PR TITLE
Feature/cb2 6899

### PR DESCRIPTION
**Pull request by mcadenas-bjss**
## Archive Tech Record still works with multiple VINs

VTM currently assumes a single Tech Record contains all information on a specific vehicle but when we change the VIN this is no longer true and so the data in VTM will no longer be accurate.

By making this change we need to ensure that all our existing backend services can still work as expected when they may have to deal with Tech Records that have had their VIN amended.
[CB2-6899](https://dvsa.atlassian.net/browse/CB2-6899)

## Checklist

- [ ] Code has been tested manually
- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/domain/Processors/VehicleProcessor.ts [hunk 1]
```diff
@@ -185,10 +205,16 @@ export abstract class VehicleProcessor<T extends Vehicle> {
...
       );
 
-      this.vehicle.techRecord[0].updateType = enums.UPDATE_TYPE.TECH_RECORD_UPDATE;
+      this.vehicle.techRecord[0].updateType =
+        enums.UPDATE_TYPE.TECH_RECORD_UPDATE;
       uniqueRecord.techRecord.push(this.vehicle.techRecord[0]);
       await this.techRecordDAO.updateSingle(uniqueRecord);
-      const mergedRecord = vehiclesFromDB.length > 1 ? this.techRecordsListHandler.mergeRecordsWithSameSystemNumber(vehiclesFromDB) : vehiclesFromDB
+      const mergedRecord =
+        vehiclesFromDB.length > 1
```
Comment (aaron-downing) [inline src/domain/Processors/VehicleProcessor.ts:213]: linting looks a bit strange on this line

File: src/domain/Processors/VehicleProcessor.ts [hunk 2]
```diff
@@ -145,15 +153,20 @@ export abstract class VehicleProcessor<T extends Vehicle> {
...
+    findFn: (...args: any[]) => boolean
+  ): T {
     if (vehiclesFromDB.length === 1) {
-      return vehiclesFromDB[0]
+      return vehiclesFromDB[0];
     }
-    const filteredRecords = vehiclesFromDB.filter(record => record.techRecord.find(findFn))
+    const filteredRecords = vehiclesFromDB.filter((record) =>
+      record.techRecord.find(findFn)
+    );
```
Comment (tahayazici) [inline src/domain/Processors/VehicleProcessor.ts:165]: This line, why?

File: src/domain/Processors/VehicleProcessor.ts [hunk 3]
```diff
@@ -165,15 +178,22 @@ export abstract class VehicleProcessor<T extends Vehicle> {
...
+        vehiclesFromDB,
+        (techRecord) => techRecord.statusCode === enums.STATUS.CURRENT
       );
-      const uniqueRecord = this.getTechRecordToUpdate(vehiclesFromDB, techRecord => techRecord.statusCode === enums.STATUS.CURRENT);
 
-      if (uniqueRecord.techRecord.find((techRecord) => techRecord.statusCode === enums.STATUS.PROVISIONAL)
+      if (
+        uniqueRecord.techRecord.find(
+          (techRecord) => techRecord.statusCode === enums.STATUS.PROVISIONAL
+        )
```
Comment (tahayazici) [inline src/domain/Processors/VehicleProcessor.ts:196]: ???

File: src/domain/Processors/VehicleProcessor.ts [hunk 4]
```diff
@@ -185,10 +205,16 @@ export abstract class VehicleProcessor<T extends Vehicle> {
         new Date().toISOString()
       );
 
-      this.vehicle.techRecord[0].updateType = enums.UPDATE_TYPE.TECH_RECORD_UPDATE;
+      this.vehicle.techRecord[0].updateType =
+        enums.UPDATE_TYPE.TECH_RECORD_UPDATE;
```
Comment (tahayazici) [inline src/domain/Processors/VehicleProcessor.ts:209]: ...

**Review by tahayazici — COMMENTED** [review]
With all jokes aside, there is so much refactoring (unnecessary at times), that I really struggled finding the actual code changes.

I thought we had this conversation already, which I was the culprit for at the time, that we would not heavily modify the styling of the code and let the auto-linter do its thing for consistency.

File: src/domain/Processors/VehicleProcessor.ts (hunk 2 above)
Comment (mcadenas-bjss) [reply src/domain/Processors/VehicleProcessor.ts:165]: It's a reusable function @gjulien-bjss made to filter vehicle records. We both use it with a different find predicate but the functionality remains the same.

File: src/domain/Processors/VehicleProcessor.ts (hunk 1 above)
Comment (mcadenas-bjss) [reply src/domain/Processors/VehicleProcessor.ts:213]: Not sure why, it's just how the linter did it.

File: src/domain/Processors/VehicleProcessor.ts (hunk 2 above)
Comment (tahayazici) [reply src/domain/Processors/VehicleProcessor.ts:165]: No, that is fair. What I meant is what is the difference/gain of refactoring from this
```typescript
    const filteredRecords = vehiclesFromDB.filter(record => record.techRecord.find(findFn));
```
to this
```typescript
    const filteredRecords = vehiclesFromDB.filter((record) =>
      record.techRecord.find(findFn)
    );
```
?

Comment (tahayazici) [reply src/domain/Processors/VehicleProcessor.ts:165]: My comments below refer to the same thing.

**Review by Ian-Hodges — APPROVED** [review]

**Pull request closed without merging**